parsers/IL.py

```diff
@@ -88,19 +141,66 @@ def extract_price_date(soup):
 def fetch_production(
     zone_key="IL", session=None, target_datetime=None, logger=None
 ) -> dict:
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
 
-    data = fetch_all()
-    production = [float(item) for item in data]
-
-    # all mapped to unknown as there is no available breakdown
-    return {
-        "zoneKey": zone_key,
-        "datetime": arrow.now(TZ).datetime,
-        "production": {"unknown": production[0] + production[1]},
-        "source": IEC_URL,
-    }
+    data = {"zoneKey": zone_key, "datetime": arrow.now(TZ).datetime}
+
+    # fetch pv capacity from IRENA
+
+    try:
+        assert (
+            post(url=IRENA_CAPACITY_URL, json=IRENA_SOLAR_PV_QUERY).status_code == 200
+        ), "IRENA solar PV url is not accessible"
+        if target_datetime:
+            data["datetime"] = arrow.get(target_datetime).datetime
+        solar_pv = fetch_IRENA_capacity(
+            irena_url=IRENA_CAPACITY_URL, irena_query=IRENA_SOLAR_PV_QUERY
+        )
+
+    except AssertionError:
```

Comment:
This whole try catch structure makes the code complicated. 

What I would suggest: 

1. In the functions (whether for production or capacity), you check whether there is data that can be fetched. If yes then return that data if not return an empty data object, while logging an Error (and not warning) saying that that data point can't be fetched.
2. Then assemble these data objects within the `fetch_production` function.
3. (Optional) if the concatenated data objects are empty it means all individual parsers (production, capacity, ..) have failed, you can raise an exception to make the parser fail. It's Optional as we would already be returning empty data while logging an error